### ISSUE TITLE
jsx-uses-vars doesn't warn unused var h1

### ISSUE THREAD

**Issue by 59naga** [yannickcr/eslint-plugin-react#1840]
The following `h1` is ignored.

```jsx
import React from 'react';
import styled from 'styled-components';

const h1 = styled.h1`
  font-size: 1.5em;
  text-align: center;
  color: palevioletred;
`;

const App = () => (
  <h1>hello</h1>
);
```

becomes:

```js
/// ...
var App = function App() {
  return _react2.default.createElement(
    'h1',
    null,
    'hello'
  );
};
```

` h1` is not used. However, it isn't warned by `jsx-uses-vars`.

**Comment by ljharb** [comment]
`jsx-uses-vars` wouldn't be involved, because the `h1` variable isn't, in fact, used in the jsx - because component names that are lowercased are HTML elements, not custom components.

In other words, this would be handled by a core rule - `no-unused-vars` - and is not related to jsx or react at all.

It seems like you want to name it something capitalized, like `H1`?

**Comment by 59naga** [comment]
Yes. This can't happen if you follow the React naming convention.
This issue asked `jsx-uses-vars` for warning as fool-proof.

But if it is due to the `no-unused-vars` behavior, I'll close this. the importance of this issue isn't higher.

I use React's naming convention.
